OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -556,43 +556,38 @@ computeFiberEquilibrium(SimTK::State& s, bool solveForVelocity) const
     double pathSpeed = solveForVelocity ? getLengtheningSpeed(s) : 0;
     double activation = getActivation(s);
 
-    try {
-        std::pair<StatusFromEstimateMuscleFiberState,
-                  ValuesFromEstimateMuscleFiberState> result =
-            estimateMuscleFiberState(activation, pathLength, pathSpeed,
-                tol, maxIter, solveForVelocity);
-
-        switch(result.first) {
-
-        case StatusFromEstimateMuscleFiberState::Success_Converged:
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Warning_FiberAtLowerBound:
-            log_warn("Millard2012EquilibriumMuscle static solution: '{}' is "
-                   "at its minimum fiber length of {}.",
-                   getName(), result.second["fiber_length"]);
-            setActuation(s, result.second["tendon_force"]);
-            setFiberLength(s, result.second["fiber_length"]);
-            break;
-
-        case StatusFromEstimateMuscleFiberState::Failure_MaxIterationsReached:
-            // Report internal variables and throw exception.
-            std::ostringstream ss;
-            ss << "\n  Solution error " << abs(result.second["solution_error"])
-               << " exceeds tolerance of " << tol << "\n"
-               << "  Newton iterations reached limit of " << maxIter << "\n"
-               << "  Activation is " << activation << "\n"
-               << "  Fiber length is " << result.second["fiber_length"] << "\n";
-            OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, ss.str());
-            break;
-        }
+    const MuscleStateEstimate result =
+        estimateMuscleFiberState(
+            activation,
+            pathLength,
+            pathSpeed,
+            tol,
+            maxIter,
+            solveForVelocity);
+
+    if (result.status ==
+        MuscleStateEstimate::Status::Failure_MaxIterationsReached) {
+            std::ostringstream oss;
+            oss << "Internal exception encountered:\n"
```

Comment:
You do not need to state `Internal exception encountered:` when throwing an exception. A higher-level system (e.g. one that `try {} catch {}`es the exception and logs it) would be responsible for pointing out that the system encountered an internal exception (and here's the message).